packages/teleterm/src/services/config/configService.ts

```diff
@@ -1,39 +1,203 @@
-import { Config, ConfigService, ConfigServiceProvider } from './types';
+import { FileStorage } from 'teleterm/services/fileStorage';
+import { Platform } from 'teleterm/mainProcess/types';
 
-type ConfigServiceProviders = {
-  readonly [Property in keyof Config]: ConfigServiceProvider<Config[Property]>;
+import { createConfigStore } from './configStore';
+
+const createAppConfigSchema = (platform: Platform) => {
+  const defaultKeymap = getDefaultKeymap(platform);
+  const defaultFonts = getDefaultFonts(platform);
+
+  // Important: all keys except 'usageMetrics.enabled' are currently not
+  // configurable by the user, we need to better validate the input (especially
+  // fonts, where any CSS can be injected).
+  // To make a key configurable, remove `omitStoredConfigValue`.
+  return z.object({
+    'keymap.tab1': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-1'])
+    ),
+    'keymap.tab2': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-2'])
+    ),
+    'keymap.tab3': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-3'])
+    ),
+    'keymap.tab4': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-4'])
+    ),
+    'keymap.tab5': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-5'])
+    ),
+    'keymap.tab6': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-6'])
+    ),
+    'keymap.tab7': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-7'])
+    ),
+    'keymap.tab8': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-8'])
+    ),
+    'keymap.tab9': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-9'])
+    ),
+    'keymap.tabClose': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-close'])
+    ),
+    'keymap.tabNew': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-new'])
+    ),
+    'keymap.tabPrevious': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-previous'])
+    ),
+    'keymap.tabNext': omitStoredConfigValue(
+      z.string().default(defaultKeymap['tab-next'])
+    ),
+    'keymap.toggleConnections': omitStoredConfigValue(
+      z.string().default(defaultKeymap['toggle-connections'])
+    ),
+    'keymap.toggleClusters': omitStoredConfigValue(
+      z.string().default(defaultKeymap['toggle-clusters'])
+    ),
+    'keymap.toggleIdentity': omitStoredConfigValue(
+      z.string().default(defaultKeymap['toggle-identity'])
+    ),
+    'keymap.openQuickInput': omitStoredConfigValue(
+      z.string().default(defaultKeymap['open-quick-input'])
+    ),
+    'fonts.sansSerifFamily': omitStoredConfigValue(
+      z.string().default(defaultFonts['sansSerif'])
+    ),
+    'fonts.monoFamily': omitStoredConfigValue(
+      z.string().default(defaultFonts['mono'])
+    ),
+    'usageMetrics.enabled': z.boolean().default(false),
```

Comment:
I think it might be a good idea to put `usageMetrics.enabled` at the top of the schema definition since it's the only special value.